lib/rpc_calls.js

```diff
@@ -50,5 +56,56 @@ export const rpc = {
       window.web3.eth.net.isListening()
             .then(r => dispatch(updateConnection(r? 1 : 0)))
     }
+  },
+
+  fetchTxHistory() {
+    return function (dispatch, getState) {
+
+      const { address, txHistory, pendingTxs } = getState();
+      if (address) {
+        if (!txHistory) {
+          dispatchUpdateTxHistory(dispatch, address, pendingTxs);
+        } else {
+          fetchEtherscanTxHistory(address, 1, 1)
+            .then(([latestTx]) => {
+              const latestTxHash = latestTx? latestTx.hash : null;
+              const firstHistoryTxHash = txHistory[0]? txHistory[0].hash : null;
+              if (latestTxHash != firstHistoryTxHash)
+                dispatchUpdateTxHistory(dispatch, address, pendingTxs);
+            })
+          }
+      }
+    }
+  }
+}
+
+function dispatchUpdateTxHistory(dispatch, address, pendingTxs) {
+  fetchEtherscanTxHistory(address, 1, 1000)
+    .then(txHistory => {
+      dispatch(updateTxHistory(txHistory));
+      dispatch(updatePendingTxs(resolvePendingTxs(txHistory, pendingTxs)));
+    })
+}
+
+function fetchEtherscanTxHistory(address, page, offset) {
+  const url = `${process.env.ETHERSCAN_URL}&address=${address}`+(page? `&page=${page}` : ``)+(offset? `&offset=${offset}` : ``);
```

Comment:
You don't have to check for empty `page` and `offset` here.

Etherscan escapes them, so:
```
https://api-ropsten.etherscan.io/api?module=account&sort=desc&action=txlist&address=0x9c01AF64A313c80d63C51A901244CFF08e8DD4CB&page=&offset=
```
returns same results as without them.

Reply:
Done